controllers/reconcilers/kserve_networkpolicy_reconciler.go

```diff
@@ -98,47 +113,128 @@ func (r *KserveNetworkPolicyReconciler) createDesiredResource(isvc *kservev1beta
 			PolicyTypes: []v1.PolicyType{v1.PolicyTypeIngress},
 		},
 	}
-	return desiredNetworkPolicy, nil
+	return desiredNetworkPolicy
+}
+
+// allowOpenshiftIngressPolicy creates policy that grants ingress access through Openshift Routes to the services which
+// are part of a namespace under Service Mesh, but are not under Service Mesh control.
+func (r *KserveNetworkPolicyReconciler) allowOpenshiftIngressPolicy(isvc *kservev1beta1.InferenceService) *v1.NetworkPolicy {
+	return &v1.NetworkPolicy{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "NetworkPolicy",
+			APIVersion: "networking.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      openshiftIngressNetworkPolicyName,
+			Namespace: isvc.Namespace,
+			Labels:    map[string]string{"opendatahub.io/related-to": "RHOAIENG-1003"},
```

Comment:
Yeah, I thought it can help in deprecation / upgrade moving fwd. Won't harm I guess.

Reply:
These type of labels or comment on code doesn't hurt the functionality but it pollute the overall code quality. I would say we could create Jira for its deprecation in future and add it to backlog so that it doesn't miss in going forward.

Reply:
Let me give you a bit more elaborated reason for the existence of this label. At some point we will provide a better-suited solution when it comes to network policies (that's actually how we started). That would mean not creating defaults on OSSM side, but also then these will go away (as they're solely to mitigate restrictions imposed by OSSM default ones). 

Considering the upgrade path with existing Data Science Projects and with those policies already existing we should take care of removing them. One way I can think of is to delete all the instances  of these special `NetworkPolicies` during the odh-operator upgrade process. Using labels makes it very straightforward to fetch all at once. Another solution (done in #136 ) is to keep their names in the code  and ensure they're removed as part of the reconciliation loop. 

> pollute the overall code quality

Can you elaborate a bit more on this? Would like to understand how I can make it better. One thing coming to mind is to extract it to a `const`.

Reply:
If promote https://github.com/opendatahub-io/odh-model-controller/pull/136 to handling deletion of NP.

Reply:
I think we differ on how the cleanup should be handled, so I hope we can work out a consensus with other reviewers :)

Thinking of it more, the logic should be implemented here, as it's this controller that owns it. I still vouch however to **not** keep the policy names in the code but leverage the concept of a label to handle such cases in a more generic fashion. I also wonder if we do need this as part of the reconciliation loop instead, maybe a one-off call during the upgrade is enough.

The name of the label is something to discuss further, as we all know it is one of the hardest problems in software engineering :)

Reply:
I agree with @israel-hdez about not having cleanup login inside a reconcile loop, especially because IMO cleanup logic should only run once. 
For example, to clean up resources related to the deprecated monitoring stack, I added the logic to a temporary initContainer for the odh-model-controller deployment which I felt was a better way of deleting those resources. This can differ for this PR though, as the number of NetworkPolicies and what namespaces they reside in is not known at startup time - the cleanup logic would have to determine that. 
For my PR the list of resources to be deleted is already known